README.md

```diff
@@ -1,3 +1,226 @@
-# kessel-sdk-py
+# Kessel SDK for Python
+
+A Python SDK for connecting to Kessel services using gRPC.
+
+## Table of Contents
+
+- [Installation](#installation)
+- [Quick Start](#quick-start)
+- [Client Builder](#client-builder)
+- [Configuration](#configuration)
+  - [Target Server](#target-server)
+  - [Credentials](#credentials)
+  - [Keep-Alive Settings](#keep-alive-settings)
+- [Authentication](#authentication)
+- [Examples](#examples)
+- [API Reference](#api-reference)
+- [Error Handling](#error-handling)
+- [Development](#development)
+
+
+## Installation
+
+```bash
+pip install kessel-sdk
+```
+
+## Quick Start
+
+```python
+import grpc
+
+from kessel.inventory.v1beta2 import (
+    ClientBuilder, 
+    rbac, 
+    check_request_pb2,
+    resource_reference_pb2, 
+    reporter_reference_pb2
+)
+
+# Build a stub pointing at a local instance
+stub = (
+    ClientBuilder
+    .with_defaults_localhost(9000)
+    .build_inventory_stub()
+)
+
+# Build helper objects
+subject = rbac.principal_subject_for_user_id("alice", "localhost")
+resource_ref = resource_reference_pb2.ResourceReference(
+    resource_id="alice_club",
+    resource_type="group",
+    reporter=reporter_reference_pb2.ReporterReference(type="rbac"),
+)
+
+check_request = check_request_pb2.CheckRequest(
+   subject=subject,
+   relation="member",
+   object=resource_ref,
+)
+
+try:
+    check_response = stub.Check(check_request)
+    print("Check response received successfully")
+    print(check_response)
+except grpc.RpcError as e:
+    print("gRPC error occurred during Check:")
+    print(f"Code: {e.code()}")
+    print(f"Details: {e.details()}")
+```
+
+
+## Client Builder
+
+The Kessel SDK uses a builder pattern to create gRPC channels and service stubs. You should reuse the same channel and stub instances for multiple API calls, as creating
+a channel is expensive. 
+`BaseClientBuilder` provides:
+
+* **Channel and Stub caching** – reuse connections automatically
+* **Reasonable defaults** – keep-alive tuned for production & development
+* **AsyncIO support** – Switch to `grpc.aio` for asynchronous tasks by calling `with_asyncio()` on the ClientBuilder
+
+
+## Configuration
+
+### Target Server
+
+The only required value is the target address (host:port):
+
+```python
+ClientBuilder.with_defaults_insecure("localhost:9000")
+ClientBuilder.with_defaults_localhost(9000)
+ClientBuilder.with_defaults(target="kessel.example.com:443", call_creds=...)
```

Comment:
We could probably get rid of the multiple `with_defaults_*` by having a single one or even the `builder` static method and using other methods to configure other aspects as mentioned above.